sematic/ui/packages/common/src/hooks/userHooks.ts

```diff
@@ -0,0 +1,19 @@
+import { UserListPayload } from "@sematic/common/src/ApiContracts";
+import { useHttpClient } from "@sematic/common/src/hooks/httpHooks";
+import useAsync from "react-use/lib/useAsync";
+
+export type QueryParams = { [key: string]: string };
+
+export function useUsersList() {
+    const { fetch } = useHttpClient();
+    const state = useAsync(async () => {
+        const response = await fetch({
+            url: "/api/v1/users"
+        });
+        return (await response.json() as UserListPayload)["content"];
+    }, [fetch]);
+
+    const { loading: isLoading, error, value } = state;
+
+    return { isLoading, error, users: value };
```

Comment:
I think as a utility function, it should return these metrics. It can be up to the caller to decide whether to consume these signals (through destructuring the return value). The utility function should fulfill its designed responsibility instead of being tailored to the caller's demand. Since there could be other usages in the future. I think many libraries we use follow this principle.

In some cases, we observe the changes in these signals to decide when to update UI (re-rendering) even though we don't directly draw UI elements using these signals. For example, if `isloading` changes from false to true, we know that the data is refreshed. But we might not have to always render `isLoading`.

`users` should be needed. It is the actual request payload.